src/brogue/Items.c

```diff
@@ -7151,11 +7162,33 @@ void detectMagicOnItem(item *theItem) {
     }
 }
 
+void increaseMaxHealthAndHeal(creature *monst, int maxHealthIncrease) {
+    char buf[1000] = "";
+
+    int healthIncrease = (monst->info.maxHP + maxHealthIncrease) * 100 / monst->info.maxHP - 100;
+    boolean isHealed = monst->currentHP < monst->info.maxHP;
+
+    monst->info.maxHP += maxHealthIncrease;
+    heal(monst, 100, true);
+
+    if (monst == &player) {
+        sprintf(buf, "%syour maximum health increases by %i%%.", isHealed ? "you heal completely and " : "", healthIncrease);
```

Comment:
I'd prefer `snprintf` generally. It's safer for future changes (that might increase the length of the string.